diff -ur gettext-0.17.old/gettext-runtime/intl/lock.h gettext-0.17/gettext-runtime/intl/lock.h
--- gettext-0.17.old/gettext-runtime/intl/lock.h	2007-11-05 01:56:16.000000000 +0100
+++ gettext-0.17/gettext-runtime/intl/lock.h	2011-11-13 12:57:39.000000000 +0100
@@ -126,8 +126,8 @@
 #  endif
 
 #  if !PTHREAD_IN_USE_DETECTION_HARD
-#   pragma weak pthread_cancel
-#   define pthread_in_use() (pthread_cancel != NULL)
+#   pragma weak pthread_exit
+#   define pthread_in_use() (pthread_exit != NULL)
 #  endif
 
 # else
Only in gettext-0.17/gettext-runtime/intl: lock.h.orig
diff -ur gettext-0.17.old/gettext-tools/gnulib-lib/lock.h gettext-0.17/gettext-tools/gnulib-lib/lock.h
--- gettext-0.17.old/gettext-tools/gnulib-lib/lock.h	2007-11-05 02:02:31.000000000 +0100
+++ gettext-0.17/gettext-tools/gnulib-lib/lock.h	2011-11-13 12:57:41.000000000 +0100
@@ -125,8 +125,8 @@
 #  endif
 
 #  if !PTHREAD_IN_USE_DETECTION_HARD
-#   pragma weak pthread_cancel
-#   define pthread_in_use() (pthread_cancel != NULL)
+#   pragma weak pthread_exit
+#   define pthread_in_use() (pthread_exit != NULL)
 #  endif
 
 # else
Only in gettext-0.17/gettext-tools/gnulib-lib: lock.h.orig
diff -ur gettext-0.17.old/gettext-tools/gnulib-lib/tls.h gettext-0.17/gettext-tools/gnulib-lib/tls.h
--- gettext-0.17.old/gettext-tools/gnulib-lib/tls.h	2007-11-05 02:02:34.000000000 +0100
+++ gettext-0.17/gettext-tools/gnulib-lib/tls.h	2011-11-13 12:57:41.000000000 +0100
@@ -70,8 +70,8 @@
 #  endif
 
 #  if !PTHREAD_IN_USE_DETECTION_HARD
-#   pragma weak pthread_cancel
-#   define pthread_in_use() (pthread_cancel != NULL)
+#   pragma weak pthread_exit
+#   define pthread_in_use() (pthread_exit != NULL)
 #  endif
 
 # else
Only in gettext-0.17/gettext-tools/gnulib-lib: tls.h.orig
diff -ur gettext-0.17.old/gettext-tools/libgettextpo/lock.h gettext-0.17/gettext-tools/libgettextpo/lock.h
--- gettext-0.17.old/gettext-tools/libgettextpo/lock.h	2007-11-05 02:03:48.000000000 +0100
+++ gettext-0.17/gettext-tools/libgettextpo/lock.h	2011-11-13 12:57:42.000000000 +0100
@@ -125,8 +125,8 @@
 #  endif
 
 #  if !PTHREAD_IN_USE_DETECTION_HARD
-#   pragma weak pthread_cancel
-#   define pthread_in_use() (pthread_cancel != NULL)
+#   pragma weak pthread_exit
+#   define pthread_in_use() (pthread_exit != NULL)
 #  endif
 
 # else
Only in gettext-0.17/gettext-tools/libgettextpo: lock.h.orig
diff -ur gettext-0.17.old/gettext-tools/libgettextpo/tls.h gettext-0.17/gettext-tools/libgettextpo/tls.h
--- gettext-0.17.old/gettext-tools/libgettextpo/tls.h	2007-11-05 02:03:49.000000000 +0100
+++ gettext-0.17/gettext-tools/libgettextpo/tls.h	2011-11-13 12:57:42.000000000 +0100
@@ -70,8 +70,8 @@
 #  endif
 
 #  if !PTHREAD_IN_USE_DETECTION_HARD
-#   pragma weak pthread_cancel
-#   define pthread_in_use() (pthread_cancel != NULL)
+#   pragma weak pthread_exit
+#   define pthread_in_use() (pthread_exit != NULL)
 #  endif
 
 # else
Only in gettext-0.17/gettext-tools/libgettextpo: tls.h.orig
